packages/experimental/Menu/src/MenuTrigger/MenuTrigger.types.ts

```diff
@@ -1,7 +1,18 @@
-import type { IViewProps } from '@fluentui-react-native/adapters';
+import { InteractionEvent, IWithPressableOptions } from '@fluentui-react-native/interactive-hooks';
+import { ViewProps } from 'react-native';
 
 export const menuTriggerName = 'MenuTrigger';
 
-export interface MenuTriggerProps extends Omit<IViewProps, 'onPress'> {}
+export interface MenuTriggerProps extends Omit<IWithPressableOptions<ViewProps>, 'onPress'> {
+  /**
+   * A RefObject to access the IButton interface. Use this to access the public methods and properties of the component.
+   */
+  componentRef?: React.RefObject<React.Component>;
 
-export interface MenuTriggerState {}
+  onClick?: (e: InteractionEvent) => void;
+}
+
+export interface MenuTriggerState {
+  props: MenuTriggerProps;
+  isSubmenu: boolean;
```

Comment:
This means that the menu trigger will host a submenu if `isSubmenu` is true? And we need to store that to know if we want to override `openOnHover` by default?

Reply:
Yeah, the MenuTrigger will host a submenu if isSubmenu is true. This is actually needed to figure out if a submenu indicator should be shown on a MenuItem. Just knowing isSubmenu from the larger menu context isn't enough, because then all MenuItems in the submenu (including the ones in the submenu's popover) will get a submenu indicator. So the MenuItem needs to know both that it's in a submenu and that it's a trigger for a submenu.

Reply:
Could we name it something like "hasSubmenu" or "hostsSubmenu" instead? I feel like "isSubmenu" feels like a relic of when we had a separate SubmenuItem control. If there's a precedent for "isSubmenu" in Fluent web, then I guess it's fine.

Reply:
Sure. The naming scheme is from FluentUI but it's not a big deal to change things internally.